File: 은빈/파이썬/[PGS]다항식더하기.py
def solution(polynomial):
    answer = ''
    x = 0
    n = 0
    
    poly = polynomial.split()
    
    for p in poly:
        if "x" in p:
            if len(p) == 1:
                x += 1
            else:
                x += int(p[:-1])
        elif '+' not in p:
            n += int(p)
            
    if x != 0:
        if x != 1:
            answer += str(x)
        answer += "x"
    if n != 0:
        if answer != "":
            answer += " + "
        answer += str(n)
            
    
    return answer

File: 은빈/자바/[PGS]다항식더하기.java
class Solution {
    public String solution(String polynomial) {
        String answer = "";
        int x = 0;
        int n = 0;
        String xNum = "";
        String nNum = "";
        
        for(int i = 0; i < polynomial.length(); i++) {
            if (polynomial.charAt(i) == 'x') {
                if (xNum.equals("")){
                    x++;
                }else {
                    x += Integer.parseInt(xNum);
                    xNum = "";
                    nNum = "";
                }
            }
            else if (polynomial.charAt(i) == '+') {
                if(!nNum.equals("")) {
                    n += Integer.parseInt(nNum);
                    nNum = "";
                    xNum = "";
                }
            }
            else if (polynomial.charAt(i) != ' ') {
                nNum += polynomial.charAt(i);
                xNum += polynomial.charAt(i);
            }
        }
        
        if(!nNum.equals("")) {
            n += Integer.parseInt(nNum);
        }
        
        if(x != 0) {
            if (x == 1) {
                answer += 'x';
            }else {
                answer += (x + "x");
            }
        }
        if(n != 0) {
            if(!answer.equals("")) {
                answer += " + ";
            }
            answer += n;
        }
        
        return answer;
    }
}